TCMManagement/Controllers/AppointmentController.cs

```diff
@@ -1,38 +1,39 @@
-using System.Collections.Generic;
-using System.Linq;
-using System.Web.Http;
-using System.Net.Http;
-using TCMManagement.BusinessLayer;
-using TCMManagement.Models;
-using static TCMManagement.BusinessLayer.Constants;
-
-namespace TCMManagement.Controllers
-{
-    /// <summary>
-    /// All Controllers should follow CRUD sequence so it will be easier to maintain the code.
-    /// </summary>
-    public class AppointmentController : ApiController
-    {
-        private IEntityServices<Appointment> appointmentService;
-
-        public AppointmentController() {
-            appointmentService = new AppointmentService();
-        }
-
-        public AppointmentController(IEntityServices<Appointment> service)
-        {
-            appointmentService = service;
-        }
-
-        // Comment our for now, easier to test
-        // [Authorize(Roles = "Receptionist")]
-        [HttpPost]
-        public IHttpActionResult AddAppointment(Appointment a)
-        {
-            return Ok(appointmentService.CreateItem(a));
-        }
-
-        // querystring = "?person=1"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web.Http;
+using System.Net.Http;
+using TCMManagement.BusinessLayer;
+using TCMManagement.Models;
+using static TCMManagement.BusinessLayer.Constants;
+
+namespace TCMManagement.Controllers
+{
+    /// <summary>
+    /// All Controllers should follow CRUD sequence so it will be easier to maintain the code.
+    /// </summary>
+    public class AppointmentController : ApiController
+    {
+        private IEntityServices<Appointment> appointmentService;
+
+        public AppointmentController() {
+            appointmentService = new AppointmentService();
+        }
+
+        public AppointmentController(IEntityServices<Appointment> service)

```

Comment:
We need to add code :
IEntityServices<Appointment> appointmentService = new AppointmentService();
            builder.RegisterInstance(appointmentService)
                .As<IEntityServices<Appointment>>()
                .SingleInstance();

to **AutofacSetup.cs** file to make DI work.

Reply:
OK.

Reply:
Not sure why, after adding this, I always confront the exception. Will investigate the reason.

Reply:
Can you please push up you code so I can clone it to my local computer and take a look?

Reply:
I changed codes back for manual test. I will push again with the changes which you can try to reproduce the issue when you are available. I will try to figure it out too.
Exception in AutoFacSetup.cs : System.ArgumentException: 'The type 'TCMManagement.BusinessLayer.AppointmentService' is not assignable to service 'TCMManagement.BusinessLayer.IEntityServices`1[[TCMManagement.BusinessLayer.AppointmentService, TCMManagement, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]]'.'